packages/rs-drive/src/drive/batch/drive_op_batch/contract.rs

```diff
@@ -78,3 +82,19 @@ impl DriveLowLevelOperationConverter for DataContractOperationType<'_> {
         }
     }
 }
+
+impl DriveOperationWithCallback for DataContractOperationType<'_> {
+    fn callback(&self, _platform_version: &PlatformVersion) -> Option<DriveOperationCallback> {
+        match self {
+            DataContractOperationType::ApplyContract { contract, .. } => {
+                let contract_id = contract.id().to_buffer();
+
+                Some(Box::new(move |drive| {
+                    let mut drive_cache = drive.cache.write().unwrap();
+                    drive_cache.cached_contracts.remove(contract_id);
+                }))
+            }
+            _ => None,
+        }
+    }
+}
```

Comment:
This seems to be the wrong approach. There is no need for a complex callback, because there are no situations that would need it.  You can't modify more than a contract at a time, hence there is no situation where we would batch an update to a contract along with a document creation, it's just not possible.

If your goal here is to support updated contracts at the batch level even though we don't use them, callbacks are still not the answer as you have all information at the document insertion time, as you do have the current batch, and can just check it for a contract insertion matching the contract for your document.

Reply:
Not sure I understood what do you mean saying complex callback. This trait defines a fucntion that return a callback that will be called after we execute grovedb operations. We do this way because operations are itself moved to the grovedb apply function so we keep callbacks around and call them after we applied.

Please explain what do you mean: "you can't modify more than a contract at a time". You can't modify multiple contracts per state transition, but you can modify multiple contracts per block and even more this modification might depend on each other. Or you can send data contract update and documents batch transition that depends on this change. So that's why we update state and cache after each state transition.

My goal is cosistently update cache when we update state so the system will work correct in all parts even without knowing that it's happening.

Reply:
Agreed to change callbacks to enum with tasks to have compiled time typing and more efficient list of items to execute